### PR TITLE
Add test for IPython notebooks with runipy

### PR DESCRIPTION
**Pull request by joleroi**
This is a continuation of https://github.com/gammapy/gammapy/pull/350

### CODE REVIEW

**Comment by cdeil** [comment]
I think `source_catalogs.ipynb` is broken, no?
This is something that I'd like to author / rewrite.
Maybe just remove that for now? (if there's any notebook that works, you can add it of course)

**Comment by joleroi** [comment]
> This looks promising ... we probably should try this for Gammapy notebook testing:
https://github.com/computationalmodelling/nbval

I agree, I will try to make this work instead of the ``runipy`` approch you started first

File: Makefile [hunk 1]
```diff
@@ -73,7 +73,13 @@ pylint:
...
 # TODO: add test and code quality checks for `examples`
-# TODO: add test for IPython notebooks in gammapy-extra
+
+gammapy-extra: 
+	git clone https://github.com/gammapy/gammapy-extra.git
 
 doc-show:
 	open docs/_build/html/index.html
+
+test-notebooks: gammapy-extra
```
Comment (joleroi) [inline Makefile]: @cdeil Do you think it is preferable to run the test via ``python setup.py test -a '--nbval' -t <notebook>``?

If yes, this requires some extra effort due to
https://github.com/astropy/astropy/blob/master/astropy/tests/runner.py#L154

**Comment by cdeil** [comment]
@joleroi - Do you see any pros or cons for us with `runipy` or `nbval`?

runipy seems the more established / used tool (stars on Github) and might also be useful to integrate Sphinx / notebooks better (not sure).
nbval might be better if we want to combine normal test reporting and notebook testing in one report.

I think both will work and I'm happy with whatever you pick.
Just wanted to know if you see any clear advantages of one or the other.
If you do, please leave a comment here, or make en entry in the dev HOWTO.

File: Makefile (hunk 1 above)
Comment (cdeil) [inline Makefile]: No, I don't see a need to integrate notebook testing with `setup.py test`.
I think `make test-notebooks` is simpler and just fine.

**Comment by joleroi** [comment]
I liked ``nbval`` a little more because the output looks familiar from other test run with ``pytest``. But as discussed offline I don't see a clear advantage of either of the packages, so we'll continue with ``runipy`` for now.

**Comment by joleroi** [comment]
@cdeil @adonath @leajouvin We now have tests for IPython notebooks, please fix your favourite one and add it to 

https://github.com/joleroi/gammapy/blob/ace09d11e036a4d03a220eef10dd41e4417dbdd8/Makefile#L80

**Comment by cdeil** [comment]
Thanks!

**Pull request merged**